doc/function.md

```diff
@@ -1,5 +1,255 @@
+- `[in] args`: Array of JavaScript values as `napi_value` representing the
+arguments of the function.
+
+Returns a Value representing the JavaScript object returned by the function.
+
+### Call
+
+Calls a Javascript function from a native add-on.
+
+```cpp
+Value Call(napi_value recv, const std::initializer_list<napi_value>& args) const;
+```
+
+- `[in] recv`: The `this` object passed to the called function.
+- `[in] args`: Initializer list of JavaScript values as `napi_value` representing
+the arguments of the function.
+
+Returns a Value representing the JavaScript object returned by the function.
+
+### Call
+
+Calls a Javascript function from a native add-on.
+
+```cpp
+Value Call(napi_value recv, const std::vector<napi_value>& args) const;
+```
+
+- `[in] recv`: The `this` object passed to the called function.
+- `[in] args`: List of JavaScript values as `napi_value` representing the
+arguments of the function.
+
+Returns a Value representing the JavaScript object returned by the function.
+
+### Call
+
+Calls a Javascript function from a native add-on.
+
+```cpp
+Value Call(napi_value recv, size_t argc, const napi_value* args) const;
+```
+
+- `[in] recv`: The `this` object passed to the called function.
+- `[in] argc`: The number of the arguments passed to the function.
+- `[in] args`: Array of JavaScript values as `napi_value` representing the
+arguments of the function.
+
+Returns a Value representing the JavaScript object returned by the function.
+
+### MakeCallback
+
+Calls a Javascript function from a native add-on after an asynchronous operation.
+
+```cpp
+Value MakeCallback(napi_value recv, const std::initializer_list<napi_value>& args) const;
+```
+
+- `[in] recv`: The `this` object passed to the called function.
+- `[in] args`: Initializer list of JavaScript values as `napi_value` representing
+the arguments of the function.
+
+Returns a Value representing the JavaScript object returned by the function.
+
+### MakeCallback
+
+Calls a Javascript function from a native add-on after an asynchronous operation.
+
+```cpp
+Value MakeCallback(napi_value recv, const std::vector<napi_value>& args) const;
+```
+
+- `[in] recv`: The `this` object passed to the called function.
+- `[in] args`: List of JavaScript values as `napi_value` representing the
+arguments of the function.
+
+Returns a Value representing the JavaScript object returned by the function.
+
+### MakeCallback
+
+Calls a Javascript function from a native add-on after an asynchronous operation.
+
+```cpp
+Value MakeCallback(napi_value recv, size_t argc, const napi_value* args) const;
+```
+
+- `[in] recv`: The `this` object passed to the called function.
+- `[in] argc`: The number of the arguments passed to the function.
+- `[in] args`: Array of JavaScript values as `napi_value` representing the
+arguments of the function.
+
+Returns a Value representing the JavaScript object returned by the function.
+
+### New
```

Comment:
I think all the new methods should be together.

Reply:
Done.